sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageExtensions.cs

```diff
@@ -5,15 +5,23 @@
 using System.Collections.Generic;
 using System.Linq;
 using Azure.Core;
+using Azure.Core.Amqp;
 using Microsoft.Azure.Amqp;
 using Microsoft.Azure.Amqp.Framing;
 
 namespace Azure.Messaging.ServiceBus.Amqp
 {
     internal static class AmqpMessageExtensions
     {
-        public static AmqpMessage ToAmqpMessage(this ServiceBusMessage message) =>
-            AmqpMessage.Create(new Data { Value = new ArraySegment<byte>(message.Body.ToBytes().IsEmpty ? Array.Empty<byte>() : message.Body.ToBytes().ToArray()) });
+        public static AmqpMessage ToAmqpMessage(this ServiceBusMessage message)
+        {
+            BinaryData body = ((AmqpDataBody)message.AmqpMessage.Body).Data.ConvertAndFlattenData();
+            return AmqpMessage.Create(
+                new Data
+                {
+                    Value = new ArraySegment<byte>(body.ToBytes().IsEmpty ? Array.Empty<byte>() : body.ToBytes().ToArray())
```

Comment:
Is `ToBytes` constructing something? Normal convention is `ToSomething` constructs something new and `AsSomething` does not. Here, you're calling it twice if it is not empty.

Reply:
Yeah ToBytes is the replacement for the Bytes property. Normally I would agree, but since I added ToBytes I felt confident enough to just use it 😄